crates/services/block_aggregator_api/src/db/storage_or_remote_db.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{
+    block_range_response::BlockRangeResponse,
+    blocks::BlockSourceEvent,
+    db::{
+        BlockAggregatorDB,
+        remote_cache::RemoteCache,
+        storage_db::StorageDB,
+        table::{
+            Blocks,
+            Column,
+            LatestBlock,
+        },
+    },
+    result::Result,
+};
+use aws_sdk_s3::config::{
+    Credentials,
+    Region,
+};
+use fuel_core_storage::{
+    Error as StorageError,
+    StorageInspect,
+    StorageMutate,
+    kv_store::KeyValueInspect,
+    transactional::{
+        AtomicView,
+        Modifiable,
+        StorageTransaction,
+    },
+};
+use fuel_core_types::fuel_types::BlockHeight;
+use std::borrow::Cow;
+
+/// A union of a storage and a remote cache for the block aggregator. This allows both to be
+/// supported in production depending on the configuration
+pub enum StorageOrRemoteDB<R, S> {
+    Remote(RemoteCache<R>),
+    Storage(StorageDB<S>),
+}
+
+impl<R, S> StorageOrRemoteDB<R, S> {
+    pub fn new_storage(storage: S, sync_from: BlockHeight) -> Self {
+        StorageOrRemoteDB::Storage(StorageDB::new(storage, sync_from))
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_s3(
+        storage: R,
+        aws_id: &str,
+        aws_secret: &str,
+        aws_region: &str,
+        aws_bucket: &str,
+        url_base: &str,
+        aws_endpoint_url: Option<String>,
+        sync_from: BlockHeight,
+    ) -> Self {
+        let client = {
+            let mut builder = aws_sdk_s3::config::Builder::new();
+            if let Some(aws_endpoint_url) = aws_endpoint_url {
+                builder.set_endpoint_url(Some(aws_endpoint_url.clone()));
+            }
+
+            let config = builder
+                .force_path_style(true)
+                .region(Region::new(Cow::Owned(aws_region.to_string())))
```

Comment:
You should not need to manually specify a region here, I believe that can be auto-discovered using the default chain.

Instead of manually putting a region here, I think you can add `.cross_region_access_enabled(true)` which will automatically handle region changes.